### PR TITLE
ESD-1638: Validate VXC update JSON with checked helpers and a field gate

### PR DESCRIPTION
**Pull request by Phil-Browne**
The VXC update JSON builder used raw type assertions, so a misspelled key or a wrong-typed value (e.g. `--json '{"rateLimt": 500}'` or `{"rateLimit": "500"}'`) was silently dropped and the CLI happily reported success on an empty update.

- Swap every raw `rawData[...].(type)` assertion in `buildUpdateVXCRequestFromJSON` for the checked `utils.JSON*` helpers already used by the buy path, so a typo or wrong type now returns a clear error.
- Add an at-least-one-field gate: a payload that resolves to no recognized field (e.g. `{}` or only unknown keys) is rejected before any API call.
- Add tests covering a typo key, a wrong-typed flat and nested value, a wrong-typed partner config, an empty object, the flat `aEndVlan`/`bEndVlan` fallback, and a valid multi-field payload.

### CODE REVIEW

File: internal/commands/vxc/vxc_inputs_update.go [hunk 1]
```diff
@@ -198,129 +204,183 @@ var buildUpdateVXCRequestFromJSON = func(jsonStr string, jsonFilePath string) (*
...
+		return nil, fmt.Errorf("aEndConfiguration: %w", err)
+	} else if present {
+		if vlan, vlanPresent, err := utils.JSONNumber(aEndConfig, "vlan"); err != nil {
+			return nil, fmt.Errorf("aEndConfiguration.vlan: %w", err)
+		} else if vlanPresent {
 			vlanInt := int(vlan)
 			if err := validation.ValidateVLAN(vlanInt); err != nil {
 				return nil, fmt.Errorf("aEndConfiguration.vlan: %w", err)
 			}
 			req.AEndVLAN = &vlanInt
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go:238]: Good catch, fixed in cc3a5c15: added the same whole-number check used for `rateLimit`/`term` before truncating to `int`, so a fractional value now returns an error instead of being silently truncated.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 2]
```diff
@@ -198,129 +204,183 @@ var buildUpdateVXCRequestFromJSON = func(jsonStr string, jsonFilePath string) (*
...
+		return nil, fmt.Errorf("bEndConfiguration: %w", err)
+	} else if present {
+		if vlan, vlanPresent, err := utils.JSONNumber(bEndConfig, "vlan"); err != nil {
+			return nil, fmt.Errorf("bEndConfiguration.vlan: %w", err)
+		} else if vlanPresent {
 			vlanInt := int(vlan)
 			if err := validation.ValidateVLAN(vlanInt); err != nil {
 				return nil, fmt.Errorf("bEndConfiguration.vlan: %w", err)
 			}
 			req.BEndVLAN = &vlanInt
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go:269]: Good catch, fixed in cc3a5c15: added the same whole-number check used for `rateLimit`/`term` before truncating to `int`, so a fractional value now returns an error instead of being silently truncated.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 3]
```diff
@@ -198,129 +204,183 @@ var buildUpdateVXCRequestFromJSON = func(jsonStr string, jsonFilePath string) (*
...
-			}
-			req.AEndVLAN = &aEndVLANInt
+	} else if aEndVLAN, present, err := utils.JSONNumber(rawData, "aEndVlan"); err != nil {
+		return nil, fmt.Errorf("aEndVlan: %w", err)
+	} else if present {
+		aEndVLANInt := int(aEndVLAN)
+		if err := validation.ValidateVLAN(aEndVLANInt); err != nil {
+			return nil, fmt.Errorf("aEndVlan: %w", err)
 		}
+		req.AEndVLAN = &aEndVLANInt
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go:251]: Good catch, fixed in cc3a5c15: added the same whole-number check used for `rateLimit`/`term` before truncating to `int`, so a fractional value now returns an error instead of being silently truncated.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 4]
```diff
@@ -198,129 +204,183 @@ var buildUpdateVXCRequestFromJSON = func(jsonStr string, jsonFilePath string) (*
...
-			}
-			req.BEndVLAN = &bEndVLANInt
+	} else if bEndVLAN, present, err := utils.JSONNumber(rawData, "bEndVlan"); err != nil {
+		return nil, fmt.Errorf("bEndVlan: %w", err)
+	} else if present {
+		bEndVLANInt := int(bEndVLAN)
+		if err := validation.ValidateVLAN(bEndVLANInt); err != nil {
+			return nil, fmt.Errorf("bEndVlan: %w", err)
 		}
+		req.BEndVLAN = &bEndVLANInt
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go:282]: Good catch, fixed in cc3a5c15: added the same whole-number check used for `rateLimit`/`term` before truncating to `int`, so a fractional value now returns an error instead of being silently truncated.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 5]
```diff
@@ -198,129 +204,183 @@ var buildUpdateVXCRequestFromJSON = func(jsonStr string, jsonFilePath string) (*
...
 
-	if aEndInnerVLAN, ok := rawData["aEndInnerVlan"].(float64); ok {
+	if aEndInnerVLAN, present, err := utils.JSONNumber(rawData, "aEndInnerVlan"); err != nil {
+		return nil, fmt.Errorf("aEndInnerVlan: %w", err)
+	} else if present {
 		aEndInnerVLANInt := int(aEndInnerVLAN)
 		if err := validation.ValidateVXCEndInnerVLAN(aEndInnerVLANInt); err != nil {
 			return nil, fmt.Errorf("invalid aEndInnerVlan: %w", err)
 		}
 		req.AEndInnerVLAN = &aEndInnerVLANInt
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go:309]: Good catch, fixed in cc3a5c15: added the same whole-number check used for `rateLimit`/`term` before truncating to `int`, so a fractional value now returns an error instead of being silently truncated.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 6]
```diff
@@ -198,129 +204,183 @@ var buildUpdateVXCRequestFromJSON = func(jsonStr string, jsonFilePath string) (*
...
 
-	if bEndInnerVLAN, ok := rawData["bEndInnerVlan"].(float64); ok {
+	if bEndInnerVLAN, present, err := utils.JSONNumber(rawData, "bEndInnerVlan"); err != nil {
+		return nil, fmt.Errorf("bEndInnerVlan: %w", err)
+	} else if present {
 		bEndInnerVLANInt := int(bEndInnerVLAN)
 		if err := validation.ValidateVXCEndInnerVLAN(bEndInnerVLANInt); err != nil {
 			return nil, fmt.Errorf("invalid bEndInnerVlan: %w", err)
 		}
 		req.BEndInnerVLAN = &bEndInnerVLANInt
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go:323]: Good catch, fixed in cc3a5c15: added the same whole-number check used for `rateLimit`/`term` before truncating to `int`, so a fractional value now returns an error instead of being silently truncated.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 7]
```diff
@@ -198,129 +204,201 @@ var buildUpdateVXCRequestFromJSON = func(jsonStr string, jsonFilePath string) (*
...
+	if aEndPartnerConfigRaw, present, err := utils.JSONObject(rawData, "aEndPartnerConfig"); err != nil {
+		return nil, fmt.Errorf("aEndPartnerConfig: %w", err)
+	} else if present {
+		connectType, _, err := utils.JSONString(aEndPartnerConfigRaw, "connectType")
+		if err != nil {
+			return nil, fmt.Errorf("aEndPartnerConfig.connectType: %w", err)
+		}
+		if strings.ToUpper(connectType) != "VROUTER" {
 			return nil, fmt.Errorf("only VRouter partner configurations can be updated")
 		}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go]: Good catch, fixed in 101ebaa0: connectType is now checked for presence and non-empty before the VROUTER comparison, so a missing connectType returns "aEndPartnerConfig.connectType is required" (or the B-End equivalent) instead of the generic "only VRouter" message.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 8]
```diff
@@ -198,129 +204,201 @@ var buildUpdateVXCRequestFromJSON = func(jsonStr string, jsonFilePath string) (*
...
+	if bEndPartnerConfigRaw, present, err := utils.JSONObject(rawData, "bEndPartnerConfig"); err != nil {
+		return nil, fmt.Errorf("bEndPartnerConfig: %w", err)
+	} else if present {
+		connectType, _, err := utils.JSONString(bEndPartnerConfigRaw, "connectType")
+		if err != nil {
+			return nil, fmt.Errorf("bEndPartnerConfig.connectType: %w", err)
+		}
+		if strings.ToUpper(connectType) != "VROUTER" {
 			return nil, fmt.Errorf("only VRouter partner configurations can be updated")
 		}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go]: Good catch, fixed in 101ebaa0: connectType is now checked for presence and non-empty before the VROUTER comparison, so a missing connectType returns "aEndPartnerConfig.connectType is required" (or the B-End equivalent) instead of the generic "only VRouter" message.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 9]
```diff
@@ -175,8 +175,11 @@ var buildUpdateVXCRequestFromJSON = func(jsonStr string, jsonFilePath string) (*
 	}
 
 	req := &megaport.UpdateVXCRequest{}
+	fieldSet := false
 
-	if rateLimit, ok := rawData["rateLimit"].(float64); ok {
+	if rateLimit, present, err := utils.JSONNumber(rawData, "rateLimit"); err != nil {
+		return nil, fmt.Errorf("rateLimit: %w", err)
+	} else if present {
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go:182]: Fixed in d4e17e31: every error path in buildUpdateVXCRequestFromJSON now wraps its returned error with exitcodes.NewUsageError, matching the buy JSON builder's convention.

File: internal/commands/vxc/vxc_inputs_test.go [hunk 10]
```diff
@@ -1456,6 +1456,412 @@ func TestBuildUpdateVXCRequestFromJSON_WholeRateLimit(t *testing.T) {
...
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := buildUpdateVXCRequestFromJSON(tt.json, "")
+			if tt.expectedError != "" {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), tt.expectedError)
+			} else {
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_test.go:1899]: Fixed in 88c38009: the FieldCoverage table now asserts each rejected case's error unwraps to *exitcodes.CLIError with Code == exitcodes.Usage.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 11]
```diff
@@ -160,185 +161,277 @@ var buildUpdateVXCRequestFromFlags = func(cmd *cobra.Command) (*megaport.UpdateV
...
+		return nil, exitcodes.NewUsageError(fmt.Errorf("failed to parse JSON: %w", err))
 	}
 
 	req := &megaport.UpdateVXCRequest{}
+	fieldSet := false
 
-	if rateLimit, ok := rawData["rateLimit"].(float64); ok {
+	if rateLimit, present, err := utils.JSONNumber(rawData, "rateLimit"); err != nil {
+		return nil, exitcodes.NewUsageError(fmt.Errorf("rateLimit: %w", err))
+	} else if present {
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go:182]: Simplified this in 94f9b9b9 — removed the redundant field-name prefix since the JSON helper's own error message already names the field. Kept the wraps that add real path/disambiguation context (nested `.vlan`/`.connectType` paths, and the A-end/B-end VLAN validation wraps).

File: internal/commands/vxc/vxc_inputs_update.go [hunk 12]
```diff
@@ -160,185 +161,277 @@ var buildUpdateVXCRequestFromFlags = func(cmd *cobra.Command) (*megaport.UpdateV
...
-			return nil, fmt.Errorf("invalid bVnicIndex: %w", err)
+			return nil, exitcodes.NewUsageError(fmt.Errorf("invalid bVnicIndex: %w", err))
 		}
 		req.BVnicIndex = &idx
+		fieldSet = true
+	}
+
+	if !fieldSet {
+		return nil, exitcodes.NewUsageError(fmt.Errorf("at least one field must be updated"))
 	}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go]: Added in 94f9b9b9 — the "at least one field must be updated" error now appends the sorted list of unrecognized top-level keys when present, while keeping the original substring intact for backward compatibility.

File: internal/commands/vxc/vxc_inputs_test.go [hunk 13]
```diff
@@ -1456,6 +1456,415 @@ func TestBuildUpdateVXCRequestFromJSON_WholeRateLimit(t *testing.T) {
 	assert.Equal(t, 2000, *req.RateLimit)
 }
 
+func TestBuildUpdateVXCRequestFromJSON_TypoKeyIsRejected(t *testing.T) {
+	// "rateLimt" is not a recognized field, so it must not be silently
+	// dropped into an empty, successful update.
+	_, err := buildUpdateVXCRequestFromJSON(`{"rateLimt": 500}`, "")
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "at least one field must be updated")
+}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_test.go:1470]: Added in 94f9b9b9 — the typo-key test now asserts the unrecognized key appears in the message and that the error classifies as `exitcodes.Usage`.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 14]
```diff
@@ -160,185 +162,297 @@ var buildUpdateVXCRequestFromFlags = func(cmd *cobra.Command) (*megaport.UpdateV
...
+		for key := range rawData {
+			if !recognizedKeys[key] {
+				unrecognized = append(unrecognized, key)
+			}
+		}
+		if len(unrecognized) > 0 {
+			sort.Strings(unrecognized)
+			return nil, exitcodes.NewUsageError(fmt.Errorf("at least one field must be updated (unrecognized keys: %s)", strings.Join(unrecognized, ", ")))
+		}
+		return nil, exitcodes.NewUsageError(fmt.Errorf("at least one field must be updated"))
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go]: Fixed in 291864b9 — the empty-update gate now also inspects the nested aEndConfiguration/bEndConfiguration objects and appends any unrecognized keys prefixed with `aEndConfiguration.`/`bEndConfiguration.` to the error message. Added tests for both the A-end and B-end nested-typo cases.

File: internal/commands/vxc/vxc_inputs_update.go [hunk 15]
```diff
@@ -160,185 +162,310 @@ var buildUpdateVXCRequestFromFlags = func(cmd *cobra.Command) (*megaport.UpdateV
...
 
-	if bEndUID, ok := rawData["bEndUid"].(string); ok {
+	if bEndUID, present, err := utils.JSONString(rawData, "bEndUid"); err != nil {
+		return nil, exitcodes.NewUsageError(err)
+	} else if present {
 		req.BEndProductUID = &bEndUID
+		fieldSet = true
 	}
 
 	// Handle partner configurations - using direct map access
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_update.go]: Fixed in 6adc8bb3 — updated the comment, it no longer references direct map access.

File: internal/commands/vxc/vxc_inputs_test.go [hunk 16]
```diff
@@ -1456,6 +1456,427 @@ func TestBuildUpdateVXCRequestFromJSON_WholeRateLimit(t *testing.T) {
...
+	var cliErr *exitcodes.CLIError
+	require.ErrorAs(t, err, &cliErr)
+	assert.Equal(t, exitcodes.Usage, cliErr.Code)
+}
+
+func TestBuildUpdateVXCRequestFromJSON_WrongTypedValueIsRejected(t *testing.T) {
+	_, err := buildUpdateVXCRequestFromJSON(`{"rateLimit": "500"}`, "")
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "rateLimit must be a number")
+}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_test.go:1501]: Added in 6adc8bb3.

File: internal/commands/vxc/vxc_inputs_test.go [hunk 17]
```diff
@@ -1456,6 +1456,427 @@ func TestBuildUpdateVXCRequestFromJSON_WholeRateLimit(t *testing.T) {
...
+
+func TestBuildUpdateVXCRequestFromJSON_EmptyObjectIsRejected(t *testing.T) {
+	_, err := buildUpdateVXCRequestFromJSON(`{}`, "")
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "at least one field must be updated")
+}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_test.go:1510]: Added in 6adc8bb3.

File: internal/commands/vxc/vxc_inputs_test.go [hunk 18]
```diff
@@ -1456,6 +1456,427 @@ func TestBuildUpdateVXCRequestFromJSON_WholeRateLimit(t *testing.T) {
...
+	assert.Equal(t, 100, *req.AEndVLAN)
+	require.NotNil(t, req.BEndVLAN)
+	assert.Equal(t, 200, *req.BEndVLAN)
+}
+
+func TestBuildUpdateVXCRequestFromJSON_WrongTypedPartnerConfigIsRejected(t *testing.T) {
+	_, err := buildUpdateVXCRequestFromJSON(`{"aEndPartnerConfig":"not-an-object"}`, "")
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "aEndPartnerConfig must be an object")
+}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/vxc/vxc_inputs_test.go:1541]: Added in 6adc8bb3.

**Review by projectsuperba — APPROVED** [review]
